gitlabform/gitlab/users.py

```diff
@@ -17,11 +17,3 @@ def get_user_by_name(self, username, user_id=None):
         if not user_id:
             user_id = self._get_user_id(username)
         return self._make_requests_to_api("users/%s", str(user_id), "GET")
-
-    def delete_user(self, username, user_id=None):
-        if not user_id:
-            user_id = self._get_user_id(username)
-        # 404 means that the user does not exist anymore, so let's accept it for idempotency
-        return self._make_requests_to_api(
-            "users/%s", str(user_id), method="DELETE", expected_codes=[204, 404]
-        )
```

Comment:
Removed as it's only used in tests, where we can directly call `gl.users.delete()` or `user.delete()` on the create object. I added a context manager to handle the expected 4xx codes.